specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json

```diff
@@ -7335,6 +7335,29 @@
       ],
       "description": "Describes a Virtual Machine Extension."
     },
+    "VirtualMachineScaleSetVMExtension": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the extension."
+        },
```

Comment:
This should be readonly to resolve the LintDiff error

Reply:
I did see why LintDiff was throwing error but VirtualMachineScaleSetExtension also is a resource and name is not readonly. I am most likely missing something, could you explain why it is not readonly for VirtualMachineScaleSetExtension but needs to be readonly here ?

Reply:
@ArcturusZhang , also this change is present from last year end (2019-07). Do we need to update compute.json in all of the versions since my initial check in was made ? https://github.com/Azure/azure-rest-api-specs/pull/7407/files

Reply:
Well, PR #7407 the `VirtualMachineExtension` inherit from [`Resource`](https://github.com/sds6/azure-rest-api-specs/blob/a65ea6ae0c9cf694273b92b4aa92316c2d8e3e63/specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json#L9979) and its `type` `name` and `id` are all read-only.

And since the `VirtualMachineScaleSetExtension` is a resource, why we do not make it inherit from the `Resource` rather than this `SubResourceReadOnly`? Is `location` the reason?
Also for completeness - the Linter recognizes ARM resource from the extension `x-ms-resource`, therefore it is reasonable to make every model with this extension could pass the linter check, rather than making some patch on its sub-types (like in this PR)

Reply:
VirtualMachineScaleSetExtension is inherting from SubResource since unlike VirtualMachineExtension it is not a full resource. We do not want to support location/tags. 

I do not understand your other comment. What are you suggesting?

Reply:
Making this `name` property readonly will solve this issue.

But I am suggesting we introduce a new base type for this particular situation that a "not full resource" is needed, rather than using this `SubResourceReadOnly` especially we have to mend it up to meet the requirement. Based on previously [where this `SubResourceReadOnly` is used](https://github.com/Azure/azure-rest-api-specs/blob/1bc134e61838ad158947963bc3f8f59fcc3665af/specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json#L6925), it looks like that this is intended to be some return value of some resource ids.

And actually if you check the common types defined [here](https://github.com/Azure/azure-rest-api-specs/blob/988270d53d4f62f229bcbebde7c00da3175cc3a9/specification/common-types/resource-management/v2/types.json#L9), a resource is not required to have a `location` or `tags`.

Reply:
And the "full resource" you are referring is called `TrackedResource` in the common-types: https://github.com/Azure/azure-rest-api-specs/blob/988270d53d4f62f229bcbebde7c00da3175cc3a9/specification/common-types/resource-management/v2/types.json#L45
Usually it is recommended to reuse the predefined types in common-types, but since this is an old swagger, we could just focus on solving the current problem - by just making this name read-only

Reply:
@ArcturusZhang I have made the name read only. 

this change is present from last year end (2019-07). Do we need to update compute.json in all of the versions since my initial check in was made ?

Reply:
I assume you are referring to introduce the same change (change VirtualMachineExtension to VirtualMachineScaleSetsVMExtension) in some older api-version. I am OK with this change, but please open a new PR for this.

Reply:
@ArcturusZhang to clarify I am asking is it *required*? If yes, why not in the same PR ?